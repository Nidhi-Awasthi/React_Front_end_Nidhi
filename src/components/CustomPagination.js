import Button from "@material-ui/core/Button";
import { makeStyles } from "@material-ui/core/styles";
import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { apiCard } from "../api/ApiCard";
import { fetchPage } from "../reducers/Card";

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  pagination: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    marginTop: "20px",
    marginBottom: "20px",
  },
  paginationButton: {
    display: "flex",
  },
  commonButtonClass: {
    backgroundColor: "Transparent",
    backgroundRepeat: "no-repeat",
    border: "none",
    cursor: "pointer",
    overflow: "hidden",
    outline: "none",
  },
  back: {
    width: "100px",
    cursor: "pointer",
  },
  next: {
    width: "100px",
    textAlign: "right",
    cursor: "pointer",
  },
  page: {
    padding: "6px 8px",
  },
}));
const CustomPagination = (props) => {
  const classes = useStyles();
  const dispatch = useDispatch();
  const {
    currentPage,
    numOfRecordsPerPage,
    getNumOfRecordsFromApiAtOneTime,
    cardList,
    pageToFetchFromApi,
  } = useSelector((state) => state.Card);
  const { setLoading } = props;
  /*back button method */
  const goToBackPage = () => {
    dispatch(fetchPage(currentPage - 1));
  };
  /*Next button method*/
  const goToNextPage = () => {
    dispatch(fetchPage(currentPage + 1));
    if (cardList.length / numOfRecordsPerPage - (currentPage + 1) === 3) {
      setLoading(true);
      dispatch(
        apiCard(pageToFetchFromApi, getNumOfRecordsFromApiAtOneTime)
      ).then((res) => setLoading(false));
    }
  };
  return (
    <div className={classes.pagination}>
      <div className={classes.paginationButton}>
        <Button
          className={(classes.back, classes.commonButtonClass)}
          onClick={() => goToBackPage()}
          disabled={currentPage === 1}
        >
          BACK
        </Button>
        <div className={classes.page}>
          Page {currentPage} of {cardList.length / numOfRecordsPerPage}
        </div>
        <Button
          className={(classes.next, classes.commonButtonClass)}
          onClick={() => goToNextPage()}
        >
          NEXT
        </Button>
      </div>
    </div>
  );
};
export default CustomPagination;
