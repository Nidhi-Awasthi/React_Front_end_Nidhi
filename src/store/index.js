import { configureStore } from "@reduxjs/toolkit";
import thunk from "redux-thunk";
import reducers from "../reducers";

function logger({ getState }) {
  return (next) => (action) => {
    // Call the next dispatch method in the middleware chain.
    const returnValue = next(action);

    // This will likely be the action itself, unless
    // a middleware further in chain changed it.
    return returnValue;
  };
}

const rootReducer = (state, action) => {
  if (action.type === "DESTROY_SESSION") {
    state = undefined;
  }
  return reducers(state, action);
};

const store = configureStore({
  reducer: rootReducer,
  middleware: [thunk, logger],
});
export default store;
