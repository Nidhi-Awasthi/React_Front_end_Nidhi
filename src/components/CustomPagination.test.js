// customCard.test.js
import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import store from "../store";
import CustomPagination from "./CustomPagination";

test("CustomPagination", () => {
  const component = renderer.create(
    <Provider store={store}>
      <CustomPagination />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
