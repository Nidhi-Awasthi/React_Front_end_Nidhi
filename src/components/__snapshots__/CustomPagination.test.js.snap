// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomPagination 1`] = `
<div
  className="makeStyles-pagination-2"
>
  <div
    className="makeStyles-paginationButton-3"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-commonButtonClass-4 Mui-disabled Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        BACK
      </span>
    </button>
    <div
      className="makeStyles-page-7"
    >
      Page 
      1
       of 
      0
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-commonButtonClass-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        NEXT
      </span>
    </button>
  </div>
</div>
`;
